### PR TITLE
S2t6 store visitor data

### PR DESCRIPTION
**Pull request by georginaw**
Back end/node functionality to get, validate and store visitor name and company. Returns response with success state to front end/react.

### CODE REVIEW

File: Services/ValidationService.js [hunk 1]
```diff
@@ -1 +1,21 @@
...
+
+let checkLength = (data) => {
+
+    return (data.length >= 1 && data.length <= 50);
+};
+
+let checkMaxLength = (data) => {
+
+    return (data.length <= 50);
+};
```
Comment (sudokufan) [inline Services/ValidationService.js]: lot of whitespace (lines 2, 7, 12, 17)

File: Services/ValidationService.js [hunk 2]
```diff
@@ -1 +1,21 @@
- 
\ No newline at end of file
+let validateVisitorSignIn = (visitorInput) => {
+
+    return (checkInputProvided(visitorInput) && checkLength(visitorInput.forename) && checkLength(visitorInput.surname) && checkMaxLength(visitorInput.company));
```
Comment (sudokufan) [inline Services/ValidationService.js]:
```suggestion
    return (checkInputProvided(visitorInput) && checkLength(visitorInput.forename)
    && checkLength(visitorInput.surname) && checkMaxLength(visitorInput.company));
```
would press Enter halfway along as shown here to make the line more readable

File: index.js [hunk 3]
```diff
@@ -1,10 +1,13 @@
...
 const cors = require('cors');
 const jsonParser = bodyParser.json();
 const app = express();
-const port = 3000;
+const port = 3001;
 app.use(cors());
 
+app.post('/visitors', jsonParser, Controller.signInVisitor);
+
 app.listen(port, () => console.log(`app listening on port ${port}`));
```
Comment (sudokufan) [inline index.js]: probably don't need the console log on the finished version of the app

File: index.js [hunk 4]
```diff
@@ -1,10 +1,13 @@
 const express = require('express');
 const bodyParser = require('body-parser');
 const DbConnector = require('./Services/DbService'); // require db service
```
Comment (sudokufan) [inline index.js]: comment probably doesn't need to be there now?

**Review by sudokufan — COMMENTED** [review]
well done

File: Services/ValidationService.js (hunk 2 above)
Comment (georginaw) [reply Services/ValidationService.js]: Done :)

File: Services/ValidationService.js (hunk 1 above)
Comment (georginaw) [reply Services/ValidationService.js]: Done

File: index.js (hunk 4 above)
Comment (georginaw) [reply index.js]: Done/deleted

File: index.js (hunk 3 above)
Comment (georginaw) [reply index.js]: Done/deleted

**Pull request merged**